tensorflow/tensorboard
# rust: add macOS Pip package builds

Summary:
This patch adds support for macOS to the build script and the CI
workflow. Remains to be seen how to get around Gatekeeper issues.

Test Plan:

 1. Acquire access to a macOS environment.
 2. Wait for the GitHub Actions workflow to succeed.
 3. Make a new virtualenv.
 4. Download the `rustboard-macos-10.15.zip` artifact.
 5. Unzip it: `unzip rustboard-macos-10.15.zip`.
 6. Install TensorBoard: `pip install tb-nightly`. (If #4429 is not yet
    in nightlies, instead install a Pip package from head.)
 7. Install the downloaded wheel: `pip install *.whl`.
 8. Launch TensorBoard with these flags:

    ```
    tensorboard --logdir ~/tensorboard_data/mnist --bind_all \
        --load_fast --verbosity 0
    ```

    This should print a log line, “Server binary (from Python package):
    [path to venv]/tensorboard_data_server/bin/server”, and the normal
    TensorBoard UI should properly display data.

wchargin-branch: rust-tds-pip-macos
